com.microsoft.java.debug.plugin/META-INF/MANIFEST.MF

```diff
@@ -3,7 +3,9 @@ Bundle-ManifestVersion: 2
 Bundle-Name: Java Debug Server Plugin
 Bundle-SymbolicName: com.microsoft.java.debug.plugin;singleton:=true
 Bundle-Version: 0.45.0
-Bundle-RequiredExecutionEnvironment: JavaSE-11
+Export-Package: com.microsoft.java.debug.core.adapter,
+ com.microsoft.java.debug.plugin.internal
```

Comment:
The ones I've used are Constants, IDebugServer and JavaDebugServer. Ideally, yes, those could be made public to facilitate integration in java-based apps.
However, I don't think there are many of them beyond Eclipse IDE, and I found a workaround using JDT-LS to invoke the startDebugging command, and that achieves that same result.
```java
ExecuteCommandParams params = new ExecuteCommandParams("vscode.java.startDebugSession", null, null);
// no API in java debug adapter to start the Debug Adapter, using the command that's contributed to jdt-ls instead
Integer port = (Integer)WorkspaceExecuteCommandHandler.getInstance().executeCommand(params, new NullProgressMonitor());
```
So I leave it to you to decide whether to publish more API or not. Since the contract of the command is clear enough and there are low risk for it to change and it's easy enough to use; I don't think it's worth opening up some APIs if it's just for this particular case.

Reply:
Thank you for sharing your use case. The VS Code client also uses the startDebugging command to start the debug server. If that works for you, I will keep this PR open for now. Pls let me know if you have more requirements to export the packages in manifest file.